### PR TITLE
Add in Rrule check

### PR DESCRIPTION
**Pull request by HemanthDogiparthi12**

### CODE REVIEW

**Review by carnellj-genesys — APPROVED** [review]
Just some small questions and changes

File: genesyscloud/resource_genesyscloud_architect_schedules.go [hunk 1]
```diff
@@ -93,9 +96,10 @@ func ResourceArchitectSchedules() *schema.Resource {
 				ValidateDiagFunc: validateLocalDateTimes,
 			},
 			"rrule": {
-				Description: "An iCal Recurrence Rule (RRULE) string.",
-				Type:        schema.TypeString,
-				Optional:    true,
+				Description:      "An iCal Recurrence Rule (RRULE) string.",
+				Type:             schema.TypeString,
```
Comment (charliecon) [inline genesyscloud/resource_genesyscloud_architect_schedules.go:100]: I noticed there is an extra bit of info in the description on API explorer: "An iCal Recurrence Rule (RRULE) string. It is required to be set for schedules determining when upgrades to the Edge software can be applied." Probably no harm to add it in

File: genesyscloud/tfexporter/export_common.go [hunk 2]
```diff
@@ -219,6 +220,40 @@ func sanitizeE164Number(number string) string {
 	}
 	return number
 }
+func sanitizeRrule(input string) string {
+	attributeRegex := map[string]*regexp.Regexp{
```
Comment (charliecon) [inline genesyscloud/tfexporter/export_common.go:224]: unit tests for this function could be good to make sure the reg expressions work as expected

File: genesyscloud/resource_exporter/resource_exporter.go [hunk 3]
```diff
@@ -194,7 +194,19 @@ func (r *ResourceExporter) IsJsonEncodable(attribute string) bool {
...
+	}
+	return lists.ItemInSlice(attribute, values)
+}
+
+func (r *ResourceExporter) IsAttributeRrule(attribute string) bool {
+	values, exists := r.CustomValidateExports["rrule"]
+	if !exists {
+		return false
+	}
+	return lists.ItemInSlice(attribute, values)
```
Comment (charliecon) [inline genesyscloud/resource_exporter/resource_exporter.go:209]: Could add unit tests here as well

**Review by charliecon — APPROVED** [review]

File: genesyscloud/tfexporter/export_common.go (hunk 2 above)
Comment (charliecon) [reply genesyscloud/tfexporter/export_common.go:224]: My bad. I assumed that was for `removeTrailingZeros`

**Pull request merged**